### DISCUSSION TITLE
Can't position controls on canvas using binding

### DISCUSSION THREAD

**Discussion by arnirichard** [AvaloniaUI/Avalonia#10753] [Q&A] [answered]
**Problem**
I am trying to position controls on canvas. I used the code given in

https://github.com/AvaloniaUI/Avalonia/discussions/10636

**To Reproduce**

1. Clone project https://github.com/arnirichard/digital_filters_dotnet
2. Run project AvaloniaFilters
3. On the left, there is a canvas with zeros (small circle) and poles (cross)

**Expected behavior**

Zeros and poles should be positioned correctly on the unit circle canvas

**Actualy behavior**

The zeros and poles are positioned in the left top corner

**Screenshots**
![image](https://user-images.githubusercontent.com/95013472/227109085-4b5f25ce-f049-4abd-a095-51ae4894620a.png)

**Desktop (please complete the following information):**

- OS: Windows
- Version Avalonia 11.0.0-preview5"

If I try to upgrade to preview6 I can't build the project anymore.

**Code**
This code is in module/folder FilterResponse, view FilterResponseView.axaml,
The viewmodel is FilterResponseViewModel.cs 
```
<ItemsControl Items="{Binding Zeros}" Grid.Row="1">
  <ItemsControl.ItemsPanel>
	  <ItemsPanelTemplate>
		  <Canvas Width="300" Height="300"/>
	  </ItemsPanelTemplate>
  </ItemsControl.ItemsPanel>
  <ItemsControl.Styles>
	  <Style Selector="ItemsControl.items > ContentControl">
		  <Setter Property="Canvas.Bottom" Value="{Binding Bottom}" />
		  <Setter Property="Canvas.Left" Value="{Binding Left}" />
	  </Style>
  </ItemsControl.Styles>
  <ItemsControl.ItemTemplate>
	  <DataTemplate>
		  <Grid Height="10" Width="10" Canvas.Left="{Binding Left}" Canvas.Bottom="{Binding Bottom}">
			  <Ellipse StrokeThickness="2" Stroke="Blue" Width="10" Height="10" />
		  </Grid>
	  </DataTemplate>
  </ItemsControl.ItemTemplate>
  </ItemsControl>
```
```
public class CanvasItem
{
    public double Bottom { get; init; }
    public double Left { get; init; }
    public object Item { get; init; }

    public CanvasItem(double left, double bottom, object item)
    {
        Bottom = bottom;
        Left = left;
        Item = item;
    }
}
```

**Comment by kekekeks** [comment, marked as answer]
Try `ItemsControl > ContentPresenter` selector

**Comment by arnirichard** [comment]
@kekekeks Thanks, that solves the problem.

**Comment by timunie** [comment]
@arnirichard just fyi: [DevTools](https://docs.avaloniaui.net/docs/getting-started/developer-tools) are really handy when you want to inspect the visual or logical tree. Call them via `[F12]`

-------

I'm converting this issue into a Q&A b'cause I feel this is not an actual bug.
